src/radical/pilot/task_description.py

```diff
@@ -4,57 +4,74 @@
 
 import radical.utils as ru
 
-
-# ------------------------------------------------------------------------------
-# Attribute description keys
-#
-UID                    = 'uid'
-NAME                   = 'name'
-EXECUTABLE             = 'executable'
-ARGUMENTS              = 'arguments'
-ENVIRONMENT            = 'environment'
-NAMED_ENV              = 'named_env'
-SANDBOX                = 'sandbox'
-
-CPU_PROCESSES          = 'cpu_processes'
-CPU_PROCESS_TYPE       = 'cpu_process_type'
-CPU_THREADS            = 'cpu_threads'
-CPU_THREAD_TYPE        = 'cpu_thread_type'
-
-GPU_PROCESSES          = 'gpu_processes'
-GPU_PROCESS_TYPE       = 'gpu_process_type'
-GPU_THREADS            = 'gpu_threads'
-GPU_THREAD_TYPE        = 'gpu_thread_type'
-
-LFS_PER_PROCESS        = 'lfs_per_process'
-MEM_PER_PROCESS        = 'mem_per_process'
-
-INPUT_STAGING          = 'input_staging'
-OUTPUT_STAGING         = 'output_staging'
-STAGE_ON_ERROR         = 'stage_on_error'
-PRE_LAUNCH             = 'pre_launch'
-PRE_EXEC               = 'pre_exec'
-PRE_RANK               = 'pre_rank'
-POST_LAUNCH            = 'post_launch'
-POST_EXEC              = 'post_exec'
-POST_RANK              = 'post_rank'
-KERNEL                 = 'kernel'
-CLEANUP                = 'cleanup'
-PILOT                  = 'pilot'
-STDOUT                 = 'stdout'
-STDERR                 = 'stderr'
-RESTARTABLE            = 'restartable'
-SCHEDULER              = 'scheduler'
-TAGS                   = 'tags'
-METADATA               = 'metadata'
+# task modules
+RP_EXECUTABLE    = 'executable'
+RP_FUNCTION      = 'function'
+RP_EVAL          = 'eval'
+RP_EXEC          = 'exec'
+RP_PROC          = 'proc'
+RP_SHELL         = 'shell'
```

Comment:
I wanted to avoid collision of the mode `executable` with the attribute `executable`.  But yeah, I don't like it much either.  On the upside, this goes away once we have different description types (`FunctionDescription`, `ExecutableDescription`, etc).  Or whatever the names will be...  But yes, renaming to `TASK_` works.